### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by cvalencia1991**
# in this milestone, I did the following activities 

- add `Book`, `Rental` and `classroom` Classes 
- implement associations with person Book and Rental 
- fix Rubocop Linters

### CODE REVIEW

File: rental.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require './book'
+require './person'
+
+class Rental
+  attr_accessor :date, :book, :person
+
+  def initialize(date, person)
+    @date = date
+    @person = person
+  end
```
Comment (mugishasam123) [inline rental.rb]: - [ ] Please try to add more explicit relations like A rental belongs to a book and also belongs to a person:

you can add those snippets in initialize method of rental class to implement that relationships:

````ruby
 # *The rental is adding itself to the @book achieving the 'belongs-to' association
    book.rentals << self
````
````ruby
    # *The rental is adding itself to the @person the 'belongs-to' association
    person.rentals << self
````

File: student.rb [hunk 2]
```diff
@@ -1,6 +1,9 @@
 require './person'
+require './classroom'
 
 class Student < Person
+  attr_accessor :classroom
+
   def initialize(*person, classroom)
```
Comment (mugishasam123) [inline student.rb:7]: - [ ] Please make sure when  setting the classroom for a student it also adds it to the classrooms' students:

you can try that code snippet in initialize  method:
``classroom.students.push(self) unless classroom.students.include?(self)``

**Review by mugishasam123 — CHANGES_REQUESTED** [review]
Hi @cvalencia1991  👋 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
alt
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- No linter errors 👍 
- Descriptive PR, keep rocking 🚀 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: rental.rb (hunk 1 above)
Comment (cvalencia1991) [reply rental.rb]: sorry but I don't understand this, I add this into the class and create the methods on it?

File: student.rb (hunk 2 above)
Comment (cvalencia1991) [reply student.rb:7]: ok i add this snippet to the student class

File: rental.rb (hunk 1 above)
Comment (cvalencia1991) [reply rental.rb]: ok, I added this relation to the rental.rb file

**Review by mahbubul14 — APPROVED** [review]
# Project Approved :trophy: :tada: 

Hi @cvalencia1991, 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linter error.
:heavy_check_mark: Descriptive readme file.
:heavy_check_mark: Well-implemented suggestions.

### Optional suggestions :arrow_down_small: 

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

- [ ] The commit message should be written in the imperative tense and the first letter should also be capitalized. `Ex: "Fix bug" instead of "fix bug".` [For more details...](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by cvalencia1991** [comment]
thank you very much @mahbubul14 😄

**Pull request merged**